dm/worker/subtask_test.go

```diff
@@ -155,6 +150,12 @@ func (t *testSubTask) TestSubTaskNormalUsage(c *C) {
 	st := NewSubTask(cfg)
 	c.Assert(st.Stage(), DeepEquals, pb.Stage_New)
 
+	// test empty and fail
+	st.units = nil
+	st.Run()
+	c.Assert(st.Stage(), Equals, pb.Stage_Paused)
+	c.Assert(strings.Contains(st.Result().Errors[0].Msg, "has no dm units for mode"), IsTrue)
+
```

Comment:
btw I found `TestSubTaskNormalUsage` sometime fails in L253, but it is not relevant to code in this pr.

Reply:
can you provide log of fail case?

Reply:
```
2019/05/28 17:20:20.771 subtask.go:411: [info] [subtask] testSubtaskScene paused with Load dm-unit
2019/05/28 17:20:20.771 subtask.go:437: [info] [subtask] testSubtaskScene resuming with Load dm-unit
2019/05/28 17:20:20.771 subtask.go:411: [info] [subtask] testSubtaskScene paused with Load dm-unit
2019/05/28 17:20:20.771 subtask.go:437: [info] [subtask] testSubtaskScene resuming with Load dm-unit

----------------------------------------------------------------------
FAIL: subtask_test.go:144: testSubTask.TestSubTaskNormalUsage

subtask_test.go:253:
    c.Assert(st.Stage(), Equals, pb.Stage_Finished)
... obtained pb.Stage = 2 ("Running")
... expected pb.Stage = 5 ("Finished")

2019/05/28 17:20:22.867 mydumper.go:241: [info] [mydumper] create mydumper using args [--host  --port 0 --user  --outputdir ]
2019/05/28 17:20:22.867 subtask.go:182: [info] [subtask] testSubtaskScene start running Dump dm-unit
2019/05/28 17:20:22.867 subtask.go:411: [info] [subtask] testSubtaskScene paused with Dump dm-unit
```

Reply:
Confirmed that the error is caused by background routine `Process` of MockUnit.
in subtask_test, each time we pause a subtask, the `Process` routine of MockUnit doesn't exit, so there exist multiple `ProcessResult` receiving process result.

Reply:
you're very right!  But I think to fix the status machine later, it has many problems